src/app/shared/directives/if-view-mode.directive.ts

```diff
@@ -0,0 +1,41 @@
+import { Directive, Input, OnDestroy, TemplateRef, ViewContainerRef } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import { ProductsView } from 'src/app/shared/models/products-view';
+import * as ProductsViewSelectors from 'src/app/shared/store/products-view/products-view.selectors';
+
+@Directive({
+  selector: '[tkIfViewMode]'
+})
+export class IfViewModeDirective implements OnDestroy {
```

Comment:
This is overengineering, please remove this directive and use simple if-else condition where needed.